completion_aggregator/signals.py

```diff
@@ -56,7 +53,7 @@ def item_deleted_handler(usage_key, user_id, **kwargs):
     # only fired from split-mongo, so it will always contain the course run.
     course_key = usage_key.course_key
     for user in get_active_users(course_key):
-        update_aggregators.delay(username=user.username, course_key=six.text_type(course_key), force=True)
+        StaleCompletion.objects.create(username=user.username, course_key=course_key, force=True)
```

Comment:
Does it make sense to add a waffle switch to decide whether to process completions with the original celery task, or to mark them Stale and process in batch?

Reply:
I have mixed feelings.  

* A separate task for every single completion could get to be a lot of tasks very quickly, especially given that you might see three or four completable-by-viewing blocks on a single screen, and submit them all within a single second, and that's for each user.  
* On the other hand, submitting the tasks directly could simplify the infrastructure for small installations that might only see a dozen users a day.
* On the third hand, having a {cron job / celerybeat task} that needs to be installed in some setups, but not in others is just pushing the complexity from one place to another.  
* And if it's a waffle switch then to activate it, you would need to both flip the switch *and* manually start up the {cron job / celerybeat task} (or have the {cron job / celerybeat task} lying around doing nothing useful when it's inactive).

I think overall making that change just adds new ways to break the system, because you could get into a state where you are creating StaleCompletions but haven't activated the {cron job / celerybeat task} to act on them.

Is there another benefit that I'm missing?

Reply:
Sure, I hear you..  if the old `task.delay` behaviour is only performant for very small instances, then it doesn't make sense to keep that.

When I looked at the celery docs, I thought we could use the code-driven [add_periodic_task](http://docs.celeryproject.org/en/latest/reference/celery.html#celery.Celery.add_periodic_task), but since we're on [celery 3.1](http://docs.celeryproject.org/en/3.1/reference/celery.html), we don't have that available.  So will have to use a supervisord or cron.

Reply:
Note that [we're still creating a celery task for each aggregator here](https://github.com/open-craft/openedx-completion-aggregator/pull/10/files/72b0ebf17ea57c618f8b0f4d612a2f498cb70139#diff-acd18b96a4b151c8e0f95ae5b0f0da71R161), it's just happening in the management command instead.  So if that's a concern, then maybe call the underlying method directly instead of using `.delay()`?